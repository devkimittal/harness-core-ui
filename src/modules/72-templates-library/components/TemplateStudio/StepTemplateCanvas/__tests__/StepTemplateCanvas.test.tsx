/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { StepTemplateCanvasWithRef } from '@templates-library/components/TemplateStudio/StepTemplateCanvas/StepTemplateCanvas'

jest.mock(
  '@templates-library/components/TemplateStudio/StepTemplateCanvas/StepTemplateDiagram/StepTemplateDiagram',
  () => ({
    ...(jest.requireActual(
      '@templates-library/components/TemplateStudio/StepTemplateCanvas/StepTemplateDiagram/StepTemplateDiagram'
    ) as any),
    StepTemplateDiagram: () => {
      return <div className="step-template-diagram-mock" />
    }
  })
)

jest.mock('@templates-library/components/TemplateStudio/StepTemplateCanvas/StepTemplateForm/StepTemplateForm', () => ({
  ...(jest.requireActual(
    '@templates-library/components/TemplateStudio/StepTemplateCanvas/StepTemplateForm/StepTemplateForm'
  ) as any),
  StepTemplateFormWithRef: () => {
    return <div className="step-template-form-mock" />
  }
}))

describe('<StepTemplateCanvas /> test', () => {
  test('Should match snapshot', () => {
    const { container } = render(<StepTemplateCanvasWithRef />)
    expect(container).toMatchSnapshot()
    const resizer = container.querySelector('[class*="Resizer"]')
    expect(resizer).toBeTruthy()
  })
})
