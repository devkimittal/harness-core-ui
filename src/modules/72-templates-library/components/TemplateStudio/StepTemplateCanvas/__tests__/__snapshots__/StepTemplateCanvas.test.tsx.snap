// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepTemplateCanvas /> test Should match snapshot 1`] = `
<div>
  <div
    class="SplitPane  horizontal "
    style="display: flex; flex: 1; height: 100%; position: absolute; outline: none; overflow: auto; user-select: text; bottom: 0px; flex-direction: column; min-height: 100%; top: 0px; width: 100%;"
  >
    <div
      class="Pane horizontal Pane1  "
      style="flex: 0 0 auto; position: relative; outline: none; height: 200px; display: flex;"
    >
      <div
        class="step-template-diagram-mock"
      />
    </div>
    <span
      class="Resizer horizontal "
      role="presentation"
      style="display: inline-table;"
    />
    <div
      class="Pane horizontal Pane2  "
      style="flex: 1; position: relative; outline: none; overflow: initial; z-index: 2;"
    >
      <div
        class="step-template-form-mock"
      />
    </div>
  </div>
</div>
`;
